code_search/code-search.ipynb

```diff
@@ -0,0 +1,509 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "source": [
+    "# Code Search on Kubeflow\n",
+    "\n",
+    "This notebook implements an end-to-end Semantic Code Search on top of [Kubeflow](https://www.kubeflow.org/) - given an input query string, get a list of code snippets semantically similar to the query string."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "\n",
+    "**NOTE**: If using the JupyterHub Spawner on a Kubeflow cluster, use the Docker image `gcr.io/kubeflow-images-public/kubeflow-codelab-notebook:v20180808-v0.2-22-gcfdcb12` which has baked all the pre-prequisites.\n",
+    "\n",
+    "* `Kubeflow v0.2.2`\n",
+    "  This notebook assumes a Kubeflow cluster is already deployed. See [Getting Started with Kubeflow](https://www.kubeflow.org/docs/started/getting-started/).\n",
+    "\n",
+    "* `Python 2.7` (bundled with `pip`) \n",
+    "  For this demo, we will use Python 2.7. This restriction is due to [Apache Beam](https://beam.apache.org/), which\n",
+    "  does not support Python 3 yet (See [BEAM-1251](https://issues.apache.org/jira/browse/BEAM-1251)).\n",
+    "\n",
+    "* `Google Cloud SDK`\n",
+    "  This example will use tools from the [Google Cloud SDK](https://cloud.google.com/sdk/). The SDK must be\n",
+    "  authenticated and authorized. See [Authentication Overview](https://cloud.google.com/docs/authentication/).\n",
+    "  \n",
+    "* `Ksonnet 0.12`\n",
+    "  We use [Ksonnet](https://ksonnet.io/) to write Kubernetes jobs in a declarative manner to be run on top of Kubeflow."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "echo \"Pip Version Info: \" && pip2 --version && echo\n",
+    "echo \"Google Cloud SDK Info: \" && gcloud --version && echo\n",
+    "echo \"Ksonnet Version Info: \" && ks version && echo\n",
+    "echo \"Kubectl Version Info: \" && kubectl version"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get the Source Code\n",
```

Comment:
It doesn't make sense to me that this is in the notebook. How did the user get code-search.ipynb if they didn't already do this step?

The flow should be something like the following.

In README.md tell the user to launch jupyter using the appropriate GCR image. Then they should start a shell via Jupyter and run git clone.

They can then use Jupyter to open up code_lab.ipynb

At which point they've already checked out kubeflow/examples.